src/app/getting-started/services/provider.service.ts

```diff
@@ -23,16 +28,43 @@ export class ProviderService {
    * @param redirect URL to be redirected to after successful account linking
    */
   linkAll(redirect: string): void {
-    this.link(null, redirect);
+        let openShiftLinkingRedirectUrl = this.getLegacyLinkingUrl("openshift-v3",redirect);
+        // after linking github, proceed with linking openshift-v3,
+        // hence passing openshift linking url as a redirect.
+        this.linkGitHub(openShiftLinkingRedirectUrl);
```

Comment:
It appears that linkAll no longer links OpenShift accounts as originally intended. And as described by the method's comments above. FYI, a null value defaulted to linking both.

If this is no longer linking both OpenShift and GitHub, perhaps it should be removed or renamed? I believe there is still logic in the getting started component that links OS, GitHub, or both at the same time.

Reply:
It does link OpenShift by passing openShiftLinkingRedirectUrl as a redirect to the GitHub API.
We have to use both old and new API to link all accounts. OSO is using the old one and GitHub the new one. So we if we need to link both OSO and GH we call the new API to link GH but we use the old API link as a redirect.

New API links GH and then redirects to the old API to link OSO. This is why we in this method there is only one call. The rest will be done by backend and both OSO and GH will be linked.